### PR TITLE
Add PHP 8.1 first class callable support

### LINKED ISSUE #936: PHP 8.1 - First class callable support
I am using prettier PHP plugin in latest version which uses the beta7 version releases yesterday and most things work fine now but first class callable support is still missing.

Examples:
```php
array_map(trim(...), $names);
```

```php
array_map($this->foo(...), $values);
```

### PR DESCRIPTION
**Pull request by MaartenStaa**
Fixes #936

### CODE REVIEW

File: test/snapshot/__snapshots__/function.test.js.snap [hunk 1]
```diff
@@ -189,6 +189,128 @@ Program {
...
+        "kind": "assign",
+        "left": Variable {
+          "curly": false,
+          "kind": "variable",
+          "name": "callable",
+        },
+        "operator": "=",
+        "right": Call {
+          "arguments": Array [
+            229,
```
Comment (czosel) [inline test/snapshot/__snapshots__/function.test.js.snap:229]: this represents the `...` (ellipsis), right? The token codes are usually not part of our output - do you know how the official PHP parser represents this?

**Review by czosel — COMMENTED** [review]
Looks great already! Just one question.

File: test/snapshot/__snapshots__/function.test.js.snap (hunk 1 above)
Comment (MaartenStaa) [reply test/snapshot/__snapshots__/function.test.js.snap:229]: Correct, that's the value of `T_ELLIPSIS`, fair point about the token codes not usually being used in the AST output. I did check how nikic/php-parser represents this, and it looks like this:

```
array(
    0: Stmt_Expression(
        expr: Expr_Assign(
            var: Expr_Variable(
                name: x
            )
            expr: Expr_FuncCall(
                name: Name(
                    parts: array(
                        0: test
                    )
                )
                args: array(
                    0: VariadicPlaceholder(
                    )
                )
            )
        )
    )
)
```

So then maybe we want a `VariadicPlaceholder` AST node. What do you think?

Comment (czosel) [reply test/snapshot/__snapshots__/function.test.js.snap:229]: Sounds good! :+1:

**Review by czosel — APPROVED** [review]
Looks great! 👍

**Pull request merged**